src/robot/dial.ts

```diff
@@ -185,3 +193,50 @@ export const createRobotClient = async (
 
   return client;
 };
+
+/**
+ * Get a Viam Transport Factory after getting the accessToken.
+ *
+ * In dialOpts.credentials, the credential type cannot be a robot secret. The
+ * credential type to use would preferably be the organization api key.
+ */
+export const createViamTransportFactory = async (
+  serviceHost: string,
+  dialOpts: DialOptions
+): Promise<grpc.TransportFactory> => {
+  if (!dialOpts.credentials) {
+    throw new Error(`credential cannot be none`);
+  } else if (dialOpts.credentials.type === 'robot-secret') {
+    throw new Error(`credential type cannot be 'robot secret'`);
```

Comment:
A more informative error message could like look:

```suggestion
    throw new Error(`credential type cannot be 'robot-secret'. Must be either 'robot-location-secret' or 'api-key'.`);
```